float32 vx
float32 vy
float32 wz
float32 psi
float32 x
float32 y